src/config/config-filter.c

```diff
@@ -38,17 +38,25 @@ static bool config_filter_match_rest(const struct config_filter *mask,
 		if (filter->local_name == NULL)
 			return FALSE;
 		/* Handle multiple names seperated by spaces in local_name
-		 * Ex: local_name "mail.domain.tld domain.tld mx.domain.tld" { ... } */
-		const char *const *local_name = t_strsplit_spaces(mask->local_name, " ");
-		bool matched = FALSE;
-		for (; *local_name != NULL; local_name++) {
-			if (dns_match_wildcard(filter->local_name, *local_name) == 0) {
-				matched = TRUE;
-				break;
+		 * Ex: local_name "mail.domain.tld domain.tld mx.domain.tld" { ... }
+		 * We check to see if the string has a " " in it so we can avoid
+		 * memory pool allocation in t_strsplit_spaces
+		 */
+		if (strstr(mask->local_name, " ") != NULL) {
```

Comment:
I'm happy to take it out, however it really is making a noticeable difference

`Test #1 compiled without ststr
# time doveconf 

real	0m11.877s
user	0m11.528s
sys	0m0.321s

Time #2 compiled with ststr
# time doveconf 
real	0m6.438s
user	0m6.090s
sys	0m0.329s

# grep local_name /etc/dovecot/sni.conf | wc
   4825   14475  161207`